iron-form.html

```diff
@@ -11,503 +11,459 @@
+       * will give you unexpected results.
+       * @return {Object} An object containing name-value pairs for elements that
+       *                  would be submitted.
+       */
+      serializeForm: function() {
+        // Only elements that have a `name` and are not disabled are submittable.
+        var elements = this._getSubmittableElements();
+        var json = {};
+        for (var i = 0; i < elements.length; i++) {
+          var values = this._serializeElementValues(elements[i]);
+          for (var v = 0; v < values.length; v++) {
+            this._addSerializedElement(json, elements[i].name, values[v]);
+          }
+        }
+        return json;
+      },
 
-    _handleFormError: function (event) {
-      this.fire('iron-form-error', event.detail);
-    },
+      _handleFormResponse: function (event) {
+        this.fire('iron-form-response', event.detail);
+      },
 
-    _registerElement: function(e) {
-      // Get the actual element that fired the event
-      var element = Polymer.dom(e).rootTarget;
+      _handleFormError: function (event) {
+        this.fire('iron-form-error', event.detail);
+      },
 
-      element._parentForm = this;
-      this._customElements.push(element);
+      _makeAjaxRequest: function(json) {
+        // Initialize the iron-ajax element if we haven't already.
+        if (!this.request) {
+          this.request = document.createElement('iron-ajax');
+          this.request.addEventListener('response', this._handleFormResponse.bind(this));
+          this.request.addEventListener('error', this._handleFormError.bind(this));
+        }
 
-      // Save the original value of this input.
-      this._customElementsInitialValues.push(
-          this._usesCheckedInsteadOfValue(element) ? element.checked : element.value);
-    },
+        // Native forms can also index elements magically by their name (can't make
+        // this up if I tried) so we need to get the correct attributes, not the
+        // elements with those names.
+        this.request.url = this._form.getAttribute('action');
+        this.request.method = this._form.getAttribute('method') || 'GET';
+        this.request.contentType = this._form.getAttribute('enctype');
+        this.request.withCredentials = this.withCredentials;
+        this.request.headers = this.headers;
+
+        if (this._form.method.toUpperCase() === 'POST') {
+          this.request.body = json;
+        } else {
+          this.request.params = json;
+        }
 
-    _unregisterElement: function(e) {
-      var target = e.detail.target;
-      if (target) {
-        var index = this._customElements.indexOf(target);
-        if (index > -1) {
-          this._customElements.splice(index, 1);
-          this._customElementsInitialValues.splice(index, 1);
+        // Allow for a presubmit hook
+        var event = this.fire('iron-form-presubmit', {}, {cancelable: true});
```

Comment:
@notwaldorf @e111077 Should it bubble?

Also, since we are creating a iron-ajax we could set the bubbles property?